src/blueapi/core/bluesky_types.py

```diff
@@ -76,7 +78,10 @@ def _follows_bluesky_protocols(obj: Any) -> bool:
 
 def is_bluesky_plan_generator(func: PlanGenerator) -> bool:
     try:
-        return get_type_hints(func).get("return") == MsgGenerator
+        return_type = get_type_hints(func).get("return")
+        return (get_origin(return_type) is Generator) and (
+            get_args(return_type)[0] is Msg
```

Comment:
`is` vs `==`? I can't remember which is recommended for comparing types, but wasn't the previous version changed to ` == MsgGenerator` to fix some bug?

Reply:
Yeah, I don't think we ever got to the bottom of that bug (#643). I'll change it and add a test case that tries to replicate it, to be safe.